src/engine/strat_engine/backstore/metadata.rs

```diff
@@ -36,7 +36,54 @@ pub struct BDA {
     regions: mda::MDARegions,
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum MetadataLocation {
+    Both,
+    First,
+    Second,
+}
+
 impl BDA {
+    /// Read the BDA from the device and returning a byte array of the entire area.
+    fn read<F>(f: &mut F) -> io::Result<[u8; _BDA_STATIC_HDR_SIZE]>
+    where
+        F: Read + Seek,
+    {
+        #![allow(unused_io_amount)]
+        f.seek(SeekFrom::Start(0))?;
+        let mut buf = [0u8; _BDA_STATIC_HDR_SIZE];
+
+        // FIXME: See https://github.com/stratis-storage/stratisd/pull/476
+        f.read(&mut buf)?;
+        Ok(buf)
+    }
+
+    /// Write the BDA, allowing the caller to specify which part(s) to update.
+    fn write<F>(f: &mut F, bda_buf: &[u8], which: MetadataLocation) -> io::Result<()>
+    where
+        F: Seek + SyncAll,
+    {
+        let zeroed = [0u8; _BDA_STATIC_HDR_SIZE];
+
+        // Write 8K header. Static_Header copies go in sectors 1 and 9.
+        if which == MetadataLocation::Both || which == MetadataLocation::First {
+            f.seek(SeekFrom::Start(0))?;
+            f.write_all(&zeroed[..SECTOR_SIZE])?; // Zero 1 unused sector
+            f.write_all(bda_buf)?;
+            f.write_all(&zeroed[..SECTOR_SIZE * 7])?; // Zero 7 unused sectors
+            f.sync_all()?;
+        } else {
+            f.seek(SeekFrom::Start((9 * SECTOR_SIZE as u64) as u64))?;
```

Comment:
The second cast here is redundant. Just
```
f.seek(SeekFrom::Start(9 * SECTOR_SIZE as u64))?;
```

will do it.

Reply:
Done indirectly with utilizing re-write of `write`